src/common/libflux/message.c

```diff
@@ -1683,11 +1683,6 @@ flux_msg_t *flux_msg_recvzsock (void *sock)
     return msg;
 }
 
-int flux_msg_frames (const flux_msg_t *msg)
-{
-    return zmsg_size (msg->zmsg);
-}
-
```

Comment:
I think this is here mainly for testing.  By removing the frame count checks from the test, are we diminishing our unit test?  Since we will still need to have a variable length stack of message frames in the rework, does it make sense to either have a function that returns a count of those, or just keep this function?  There will still be the same message frames on the wire in the rework unless we are planning to redo the protocol and RFC 3.

See zmq_send(3) - the terminology is slightly different without czmq.  zmq refers to multi-part messages, i.e. "message parts" instead of "message frames".  So maybe a rename to `flux_msg_part_count()` or similar if we decide to keep this function.

Reply:
We could just keep the function.  I was mostly thinking that counting frames suggests how data is stored internally and since we're planning on changing that, the function would have less purpose going forward.

Reply:
I think you make a good case for not exposing this in the public API.  If you plan to have a `message_private.h` in the next PR to privately share the message struct with librouter, maybe this function belongs there under a new name like `msg_multipart_count()` or similar.